### DISCUSSION TITLE
Give users sudo access

### DISCUSSION THREAD

**Discussion by dsajanice** [Azure/CloudShell#291] [Ideas]
Cloud Shell doesn't currently give users sudo access. This sometimes poses genuine challenges like not being able to install specific versions of commands. If this is something that interests you, please comment/upvote.

**Comment by brendank310** [comment]
It might be useful to scope the sudo access to a certain number of allowed utilities/apps (e.g. apt, dnf, etc) rather than giving every user the ability to sudo anything.

**Comment by edyoung** [comment]
The biggest issue is that it would allow users to modify files outside the
home directory, but those changes will be lost as soon as the container
restarts, which is almost certainly not what people expect or want.

**Reply by brendank310** [reply]
Can we create a directory in the user's home directory (let's say ~/.rootfsoverlay) and have a read-write overlay mount over the read only rootfs? Then user writes to / will persist in their overlay which is stored in their account. Something like:
`mount -t overlay overlay -olowerdir=/,upperdir=$HOME/.rootfsoverlay,workdir=$HOME/.rootfsoverlay/work /`
Curious users might mess with the mounts, but like you said that gets flushed away on the next container spinup (which while maybe not expected is the behavior today).
